hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleReadyForExecution.java

```diff
@@ -27,7 +27,17 @@
 import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
 import com.hederahashgraph.api.proto.java.ScheduleID;
 
-public abstract class ScheduleReadyForExecution {
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+
+/**
+ * Defines a class to handle scheduled transaction execution
+ * once the scheduled transaction is signed by the required
+ * number of parties.
+ *
+ * @author Michael Tinker
+ * @author Abhishek Pandey
+ */
+public class ScheduleReadyForExecution {
```

Comment:
Thanks for adding some documentation.

I see that the class is non-final, yet the documentation doesn't give me any idea as to whether this is intentional or not. I don't want to see a comment that says something like "This class is non-final so it can be extended", but rather something like "This base class does X. Subclasses should override X, Y, and Z to implement ABC". Something with more meat to it. Or, if the class is not meant to be extended, lets forbid it by making the class final.

Another thought is that you can probably run the comment out to 80 chars instead of breaking at 60 or so.

Reply:
I believe it could be made final and `ScheduleCreateTransitionLogic`, `ScheduleSignTransitionLogic` which extend from this class could use the base class instance as a member variable. Refactoring has introduced some unit tests failures for now which would be fixed with the help of @tinker-michaelj

Reply:
Refactored to use the new final class `ScheduleExecutor`, and modified the documentation comments to be around 80 chars.